### PR TITLE
Use plugin.Schema() in `wash validate`

### PR DESCRIPTION
**Pull request by ekinanp**
Signed-off-by: Enis Inan <enis.inan@puppet.com>

Contributions to this project require sign-off consistent with the [Developers Certificate of Origin](https://developercertificate.org). This can be as simple as using `git commit -s` on each commit.

### CODE REVIEW

File: cmd/validate.go [hunk 1]
```diff
@@ -289,7 +297,11 @@ func processEntry(ctx context.Context, pw progress.Writer, wp cmdutil.Pool, e pl
 					continue
 				}
 
-				ccrit := newCriteria(entry)
+				ccrit, err := newCriteria(entry)
```
Comment (MikaelSmith) [inline cmd/validate.go]: We don't need to shell out here, so I'd prefer to separate criteria as a hash object from how we stringify it (which needs more info from the schema).

Comment (ekinanp) [reply cmd/validate.go]: plugin.Schema is going to shell out if the external plugin doesn't prefetch its schemas, so not sure what you mean here.

Comment (MikaelSmith) [reply cmd/validate.go]: Didn't we separate TypeID so you don't need to call `schema` for it? That's the only one required by the hash. The label and singleton are only used when deciding how to display it, which we do once at the beginning.

Comment (ekinanp) [reply cmd/validate.go]: Oh I see what you're saying. Will update.

File: cmd/validate.go [hunk 2]
```diff
@@ -292,7 +305,7 @@ func processEntry(ctx context.Context, pw progress.Writer, wp cmdutil.Pool, e pl
 				ccrit := newCriteria(entry)
 				// If we have a TypeID, only explore children if they are different from the parent.
 				// This prevents simple recursion like volume directories containing more dirs.
-				if ccrit.typeID == "" || ccrit != crit {
+				if ccrit.typeID == "" || ccrit.typeID != crit.typeID {
```
Comment (ekinanp) [inline cmd/validate.go]: Wonder if this should be `ccrit.typeID == "" || (crit.typeID != "" && ccrit.typeID != crit.typeID)`

**Review by MikaelSmith — APPROVED** [review]

**Pull request merged**